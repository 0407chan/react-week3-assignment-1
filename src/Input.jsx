export default function Input({ value, onChange, onClick }) {
  const handleChange = (event) => {
    onChange(event.target.value);
  };

  return (
    <p>
      <label htmlFor="input-task-title">
        할 일
      </label>
      <input
        id="input-task-title"
        type="text"
        placeholder="할 일을 입력해 주세요"
        aria-label="TodoContent"
        value={value}
        onChange={handleChange}
      />
      <button type="button" onClick={onClick}>
        추가
      </button>
    </p>
  );
}
