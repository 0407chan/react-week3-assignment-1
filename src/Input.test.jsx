import { render, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Input from './Input';

test('Input 컴포넌트 구성요소는 할 일 label, input form, 추가 버튼이 있다', () => {
  const { container, getByPlaceholderText } = render(<Input />);

  expect(container).toContainHTML('label');
  expect(container).toContainHTML('input');
  expect(container).toContainHTML('button');

  expect(container).toHaveTextContent('할 일');
  getByPlaceholderText('할 일을 입력해 주세요');
  expect(container).toHaveTextContent('추가');
  expect(container.firstChild).toMatchSnapshot();
});

test('label을 선택하면, input form에 focus가 된다.', () => {
  const { container, getByText } = render(<Input />);

  userEvent.click(getByText('할 일'));
  expect(container.querySelector('input')).toHaveFocus();
});

test('input form에 할 일을 입력할 수 있다.1', () => {
  const input = render(<Input />).container.querySelector('input');

  userEvent.type(input, '오늘의 할 일');
  expect(input).toHaveValue('오늘의 할 일');
});

test('input form에 할 일을 입력할 수 있다.2', () => {
  const { getByPlaceholderText } = render(<Input />);
  const input = getByPlaceholderText('할 일을 입력해 주세요');

  fireEvent.change(input, { target: { title: '오늘의 할 일' } });
  expect(input.title).toBe('오늘의 할 일');
});

test('입력한 할 일을 추가하면 input form이 비워진다.', () => {
  const addTodo = jest.fn();
  const { getByText, getByPlaceholderText } = render(<Input onClick={addTodo} />);
  const input = getByPlaceholderText('할 일을 입력해 주세요');

  expect(addTodo).not.toBeCalled();
  userEvent.type(input, '오늘의 할 일');
  fireEvent.click(getByText('추가'));
  expect(addTodo).toBeCalled();
  expect(input.title).toBe('');
});
