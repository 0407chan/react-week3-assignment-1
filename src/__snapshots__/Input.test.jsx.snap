// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input 컴포넌트 구성요소는 할 일 label, input form, 추가 버튼이 있다 1`] = `
<p>
  <label
    for="input-task-title"
  >
    할 일
  </label>
  <input
    id="input-task-title"
    placeholder="할 일을 입력해 주세요"
    type="text"
    value=""
  />
  <button
    type="button"
  >
    추가
  </button>
</p>
`;
